src/common/users.ts

```diff
@@ -339,3 +349,56 @@ export const getUserReadingRank = async (
     tags,
   };
 };
+
+export enum Day {
+  Sunday,
+  Monday,
+  Tuesday,
+  Wednesday,
+  Thursday,
+  Friday,
+  Saturday,
+}
+
+export const Weekends = [Day.Sunday, Day.Saturday];
+export const FREEZE_DAYS_IN_A_WEEK = Weekends.length;
+export const MISSED_LIMIT = 1;
+
+export const clearUserStreak = async (
+  con: DataSource | EntityManager,
+  userId,
+): Promise<boolean> => {
+  const result = await con
+    .getRepository(UserStreak)
+    .update({ userId }, { currentStreak: 0, updatedAt: new Date() });
+
+  return result.affected > 0;
+};
+
+export const checkAndClearUserStreak = async (
+  ctx: Context,
+  info: GraphQLResolveInfo,
+  streak: GQLUserStreak,
+): Promise<boolean> => {
+  const { lastViewAt } = streak;
+
+  if (!lastViewAt) {
+    return false;
+  }
+
+  const today = new Date();
```

Comment:
We need to convert it to the user's timezone don't we?

Reply:
Yup, we do

Reply:
This comment probably got missed before :) 
https://github.com/dailydotdev/daily-api/pull/1658#discussion_r1471146921

Reply:
Still WIP with the timezones, I am still working on making the query convert the `lastViewAt` to the user preferred timezone. After that, I will match this Date instance.